mne/epochs.py

```diff
@@ -1826,8 +1827,12 @@ def __init__(self, data, info, events, tmin=0, event_id=None,
         if data.shape[0] != len(events):
             raise ValueError('The number of epochs and the number of events'
                              'must match')
-        tmax = (data.shape[2] - 1) / info['sfreq'] + tmin
-        if event_id is None:  # convert to int to make typing-checks happy
+        index_offset = 1 if data.shape[2] > 1 else 0
+        tmax = (data.shape[2] - index_offset) / info['sfreq'] + tmin
+        if index_offset == 0:
+            reject_tmin, reject_tmax = tmin, tmax
+        if event_id is None or isinstance(event_id, int):
```

Comment:
I don't think this logical should have been changed. If `event_id` is an int, you are now going to make `event_id` a dict containing *all* event numbers instead of just the given one, which is different from before. I think the super class was meant to handle that case, unless this addressing another bug simultaneously?